packages/contracts-bedrock/contracts/echidna/FuzzBurn.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity 0.8.15;
+
+import { Burn } from "../libraries/Burn.sol";
+import { StdUtils } from "forge-std/Test.sol";
+
+contract EchidnaFuzzBurnEth is StdUtils {
+    bool failedEthBurn;
+
+    /**
+     * @notice Takes an integer amount of eth to burn through the Burn library and
+     * updates the contract state if an incorrect amount of eth moved from the contract
+     */
+    function testBurn(uint256 _value) public {
+        // cache the contract's eth balance
+        uint256 preBurnBalance = address(this).balance;
+        uint256 value = bound(_value, 0, preBurnBalance);
+
+        // execute a burn of _value eth
+        Burn.eth(value);
```

Comment:
Not sure if this is easily testable here, but we *do* rely (in the actual contracts, not here) on the EVM behavior that `Burn.eth` does actually *burn* the ETH by removing it from the total supply rather than locking it into some other address.

Reply:
yeah, I considered that but don't think we can test it. 
The best we could do, afaict, would be to change this function to return the address of `Burner`, then verify that that address is empty in a future block. But I don't think that's necessary.

https://github.com/ethereum-optimism/optimism/blob/0ac0098559bb4e918491af2bbc75adf1ddf1974a/packages/contracts-bedrock/contracts/libraries/Burn.sol#L14-L16